src/routes/v3/follow.rs

```diff
@@ -0,0 +1,107 @@
+use super::ApiError;
+use crate::{
+    auth::get_user_from_headers,
+    database::{models::DatabaseError, redis::RedisPool},
+    models::pats::Scopes,
+    queue::session::AuthQueue,
+};
+use actix_web::{web, HttpRequest, HttpResponse};
+use sqlx::PgPool;
+
+use crate::database::models as db_models;
+
+pub trait HasId<T> {
+    fn id(&self) -> T;
+}
+
+impl HasId<db_models::UserId> for db_models::User {
+    fn id(&self) -> db_models::UserId {
+        self.id
+    }
+}
+
+impl HasId<db_models::OrganizationId> for db_models::Organization {
+    fn id(&self) -> db_models::OrganizationId {
+        self.id
+    }
+}
+
+pub async fn follow<T, TId, Fut1, Fut2>(
```

Comment:
this pattern also makes the code very hard to follow, I would just use the same pattern project following has, with 2 separate routes for both orgs and users, and put the code there. I understand it removes some repeated code, but it's not worth the cost esp bc nothing else will have following, this seems like you're designing a system with the intention we'll have 10+ things support following

Reply:
Yeah, this turned out much clunkier than I wanted. I disagree with the implication that something needs 10+ usages to be worth avoiding duplication, but in this case, I think the benefit is slim enough that I'll inline and duplicate them as you'd like.

This could have been done much more elegantly with traits, but unfortunately `async fn` isn't yet valid in trait definitions, so this was the best I could come up with. That should be coming with [Rust 1.75](https://releases.rs/docs/1.75.0/) in December, too bad we don't have it now.

Reply:
I was exaggerating yeah but I think it would be worth inlining these

Reply:
Done